examples/data_manager_and_text_classification.py

```diff
@@ -0,0 +1,130 @@
+
+        ax.plot(performance_history)
+        ax.scatter(range(len(performance_history)), performance_history, s=13)
+
+        ax.xaxis.set_major_locator(
+            mpl.ticker.MaxNLocator(nbins=N_QUERIES + 3, integer=True)
+        )
+        ax.xaxis.grid(True)
+
+        ax.yaxis.set_major_locator(mpl.ticker.MaxNLocator(nbins=10))
+        ax.yaxis.set_major_formatter(mpl.ticker.PercentFormatter(xmax=1))
+        ax.set_ylim(bottom=0, top=1)
+        ax.yaxis.grid(True, linestyle="--", alpha=1 / 2)
+
+        ax.set_title("Incremental classification accuracy")
+        ax.set_xlabel("Query iteration")
+        ax.set_ylabel("Classification Accuracy")
+
+        plt.show()
+
+
+if __name__ == "__main__":
+    docs_train, original_labels_train, docs_test, original_labels_test = prepare_data()
+    vectors_train, vectors_test = prepare_features(docs_train, docs_test)
+    manager = prepare_manager(vectors_train, docs_train)
+    learner = prepare_learner()
+    performance_history = []
+    # performance_history.append(learner.score(docs_test, original_labels_test))
+
+    for i in range(N_QUERIES):
+        # Check if there are more examples that are not labeled. If not, break
+        if manager.unlabeld.size == 0:
+            break
+
+        for index in range(1):
+            # query the learner as usual, in this case we are using a batch learning strategy
+            # so indices_to_label is an array
+            indices_to_label, query_instance = learner.query(manager.unlabeld)
+            labels = []  # Hold a list of the new labels
+            for ix in indices_to_label:
+                """
+                Here is the tricky part that the manager solves. The indicies are indexed with respect to unlabeled data
+                but we want to work with them with respect to the original data. The manager makes this almost transparent
+                """
+                # Map the index that is with respect to unlabeled data back to an index with respect to the whole dataset
+                original_ix = manager.get_original_index_from_unlabeled_index(ix)
+                # print(manager.sources[original_ix]) #Show the original data so we can decide what to label
+                # Now we can lookup the label in the original set of labels without any bookkeeping
+                y = original_labels_train[original_ix]
+                # We create a Label instance, a tuple of index and label
+                # The index should be with respect to the unlabeled data, the add_labels function will automatically
+                # calculate the offsets
+                label = (ix, y)
+                # append the labels to a list
+                labels.append(label)
+            # Insert them all at once.
+            manager.add_labels(labels)
+            # Note that if you need to add labels with indicies that repsect the original dataset you can do
+            # manager.add_labels(labels,offset_to_unlabeled=False)
```

Comment:
Why do you put this into a `for` block? the `index` variable is not used and the iterator which you loop through has a single element.